mantle/platform/api/azure/instance.go

```diff
@@ -37,10 +37,23 @@ type Machine struct {
 }
 
 func (a *API) getVMParameters(name, userdata, sshkey, storageAccountURI string, ip *network.PublicIPAddress, nic *network.Interface) compute.VirtualMachine {
+
+	// Azure requires that either a username/password be set or an SSH key.
+	//
+	//    Message="Authentication using either SSH or by user name and
+	//             password must be enabled in Linux profile."
+	//
+	// Since we don't ship their agent setting a username and password here
+	// is harmless. We set the username and password because some tests explicitly
+	// don't want to pass an SSH key via the API and this allows us to do that.
+
 	osProfile := compute.OSProfile{
-		AdminUsername: util.StrToPtr("core"),
+		AdminUsername: util.StrToPtr("core"),          // unused
+		AdminPassword: util.StrToPtr("ABC123&456xyz"), // unused
```

Comment:
If we ever did ship the Azure agent, or the code was adapted by someone who did, this could have unfortunate side effects.  How about randomly generating a string on every call?

Reply:
+1 - I started down this route but the password has requirements:

```
The supplied password must be between 6-
72 characters long and must satisfy at least 3 of password complexity requirements from the following:\r\n1) Contains an uppercase character\
r\n2) Contains a lowercase character\r\n3) Contains a numeric digit\r\n4) Contains a special character\r\n5) Control characters are not allow
ed"
```

So it started to get sufficiently complex that I just did a fallback to this since the password wasn't going to be used anyway. I think maybe a good middle ground is something like `ABC123${random30numbers}456xyz`. WDYT?

Reply:
see latest upload